bugsnag-android-core/src/androidTest/java/com/bugsnag/android/StacktraceTest.java

```diff
@@ -74,6 +76,19 @@ public void testStacktraceTrimming() throws Throwable {
         assertEquals(200, jsonArray.length());
     }
 
+    @Test
+    public void testStacktraceTrimmingListCtor() throws Throwable {
+        List<Map<String, Object>> elements = new ArrayList<>();
+
+        for (int k = 0; k < 1000; k++) {
+            elements.add(Collections.singletonMap("Foo", new Object()));
+        }
+
+        Stacktrace stacktrace = new Stacktrace(elements);
+        JSONArray jsonArray = streamableToJsonArray(stacktrace);
+        assertEquals(200, jsonArray.length());
+    }
```

Comment:
This test should also assert that the correct values are being truncated. The test object should have a way to differentiate each ordered frame and then later assert that the "top" 200 frames are what have been included.